pkg/kubectl/cmd/drain.go

```diff
@@ -0,0 +1,313 @@
+
+	cmdNamespace, _, err := o.factory.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	r := o.factory.NewBuilder().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		ResourceNames("node", args[0]).
+		Do()
+
+	if err = r.Err(); err != nil {
+		return err
+	}
+
+	return r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		o.nodeInfo = info
+		return nil
+	})
+}
+
+// RunDrain runs the 'drain' command
+func (o *DrainOptions) RunDrain() error {
+	if err := o.RunCordon(true); err != nil {
+		return err
+	}
+
+	pods, err := o.getPodsForDeletion()
+	if err != nil {
+		return err
+	}
+
+	if err = o.deletePods(pods); err != nil {
+		return err
+	}
+	cmdutil.PrintSuccess(o.mapper, false, o.out, "node", o.nodeInfo.Name, "drained")
+	return nil
+}
+
+// getPodsForDeletion returns all the pods we're going to delete.  If there are
+// any unmanaged pods and the user didn't pass --force, we return that list in
+// an error.
+func (o *DrainOptions) getPodsForDeletion() ([]api.Pod, error) {
+	pods := []api.Pod{}
+	podList, err := o.client.Pods(api.NamespaceAll).List(api.ListOptions{FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": o.nodeInfo.Name})})
+	if err != nil {
+		return pods, err
+	}
+	unreplicatedPodNames := []string{}
+
+	for _, pod := range podList.Items {
+		_, found := pod.ObjectMeta.Annotations[types.ConfigMirrorAnnotationKey]
+		if found {
+			// Skip mirror pod
+			continue
+		}
+		replicated := false
+
+		creatorRef, found := pod.ObjectMeta.Annotations[controller.CreatedByAnnotation]
+		if found {
+			// Now verify that the specified creator actually exists.
+			var sr api.SerializedReference
+			err := api.Scheme.DecodeInto([]byte(creatorRef), &sr)
+			if err != nil {
+				return pods, err
+			}
+			if sr.Reference.Kind == "ReplicationController" {
+				rc, err := o.client.ReplicationControllers(sr.Reference.Namespace).Get(sr.Reference.Name)
+				// Assume the only reason for an error is because the RC is
+				// gone/missing, not for any other cause.  TODO(mml): something more
+				// sophisticated than this
+				if err == nil && rc != nil {
+					replicated = true
+				}
+			} else if sr.Reference.Kind == "DaemonSet" {
+				ds, err := o.client.DaemonSets(sr.Reference.Namespace).Get(sr.Reference.Name)
+
+				// Assume the only reason for an error is because the RC is
+				// gone/missing, not for any other cause.  TODO(mml): something more
+				// sophisticated than this
+				if err == nil && ds != nil {
+					replicated = true
+				}
+			}
+		}
+		if replicated || o.Force {
+			pods = append(pods, pod)
+		}
+		if !replicated {
+			unreplicatedPodNames = append(unreplicatedPodNames, pod.Name)
+		}
+	}
+
+	if len(unreplicatedPodNames) > 0 {
+		joined := strings.Join(unreplicatedPodNames, ", ")
+		if !o.Force {
+			return pods, fmt.Errorf("Refusing to continue due to pods managed by neither a ReplicationController nor a DaemonSet: %s (use --force to override)", joined)
```

Comment:
How about "refusing to continue due to unreplicated pods". Also note that error messages need to start with a lowercase letter